tools/time_tools.go

```diff
@@ -0,0 +1,24 @@
+package tools
+
+import "time"
+
+// IsExpiredAtOrIn returns whether or not the result of calling TimeAtOrIn is
+// "expired" within "until" units of time from now.
+func IsExpiredAtOrIn(until time.Duration, at time.Time, in time.Duration) (time.Time, bool) {
+	expiration := TimeAtOrIn(at, in)
+	if expiration.IsZero() {
+		return expiration, false
+	}
+
+	return expiration, expiration.Before(time.Now().Add(until))
+}
```

Comment:
Good call, I think I addressed this in https://github.com/git-lfs/git-lfs/pull/2130/commits/0d66374f4f2da583bbbed1da621b778547175e60. I'm not a super huge fan of embedding a `createdAt` field on both of those structs, but I think it should work alright.